### ISSUE TITLE
Performance issue warming the stache on sites w/ assets in s3

### ISSUE THREAD

**Issue by edalzell** [statamic/cms#6831]
### Bug description

We have a site that uses s3 for its assets, and on .41 & .42 warming the stache takes more than 30 mins, locally. On .40 it takes 1.5 mins.

### How to reproduce

Get a site with lots of assets in s3 and run `php please stache:refresh`

### Environment

```yaml
Environment
Application Name: Statamic
Laravel Version: 9.34.0
PHP Version: 8.1.11
Composer Version: 2.4.2
Environment: local
Debug Mode: ENABLED
URL: zakat.test
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: NOT CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: daily
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 20
Antlers: runtime
Version: 3.3.40 PRO

Statamic Addons
aryehraber/statamic-captcha: 1.9.0
aryehraber/statamic-impersonator: 2.4.1
aryehraber/statamic-uuid: 2.1.0
doublethreedigital/duplicator: 2.3.2
doublethreedigital/runway: 2.5.3
edalzell/forma: 1.2
jacksleight/statamic-bard-mutator: 1.1.2
jacksleight/statamic-bard-texstyle: 1.0.3
jonassiewertsen/statamic-livewire: 2.9.0
statamic/seo-pro: 3.1.0
swiftmade/statamic-clear-assets: 1.1.0
transformstudios/events: 4.0.6
transformstudios/front: 1.10
transformstudios/gated: dev-main
transformstudios/github: dev-main
transformstudios/magic-link: dev-main
transformstudios/memorial: dev-main
transformstudios/review: 3.6.0
transformstudios/simple: 2.8.0
transformstudios/takeover-redirects: dev-main
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

runtime (new)

**Comment by nfragkos** [comment]
Same here. Running php please optimize:clear ; php please stache:warm commands on **3.3.40 takes 8 seconds**. On **v3.3.42 takes 1 minute and 20 seconds**.

**Comment by edalzell** [comment]
re-opening because the above PR is temporary

**Comment by jesseleite** [comment]
Fixed by #6871 👍
